deploy/sre-pruning/115-pruning-rename-MUO-CR.CronJob.yaml

```diff
@@ -0,0 +1,66 @@
+apiVersion: v1
+kind: ServiceAccount
+metadata:
+  name: osd-muo-cr-rename-sa
+  namespace: openshift-managed-upgrade-operator
+---
+apiVersion: rbac.authorization.k8s.io/v1
+kind: Role
+metadata:
+  name: osd-muo-cr-rename-role
+  namespace: openshift-managed-upgrade-operator
+rules:
+  - apiGroups:
+      - upgrade.managed.openshift.io
+    resources:
+      - upgradeconfigs
+    verbs:
+      - get
+      - list
+      - delete
+      - create
+      - patch
+---
+apiVersion: rbac.authorization.k8s.io/v1
+kind: RoleBinding
+metadata:
+  name: osd-muo-cr-rename-rb
+  namespace: openshift-managed-upgrade-operator
+roleRef:
+  kind: Role
+  name: osd-muo-cr-rename-role
+  apiGroup: rbac.authorization.k8s.io
+subjects:
+- kind: ServiceAccount
+  name: osd-muo-cr-rename-sa
+  namespace: openshift-managed-upgrade-operator
+---
+apiVersion: batch/v1beta1
+kind: CronJob
+metadata:
+  name: osd-muo-cr-rename
+  namespace: openshift-managed-upgrade-operator
+spec:
+  failedJobsHistoryLimit: 1
+  ttlSecondsAfterFinished: 100
+  successfulJobsHistoryLimit: 3
+  concurrencyPolicy: Replace
+  schedule: "*/10 * * * *"
+  jobTemplate:
+    spec:
+      template:
+        spec:
+          restartPolicy: Never
+          serviceAccountName: osd-muo-cr-rename-sa
+          containers:
+            - name: osd-muo-cr-rename
+              imagePullPolicy: Always
+              image: image-registry.openshift-image-registry.svc:5000/openshift/cli:latest
+              args:
+                - /bin/bash
+                - -euo pipefail -c
+                - |
+                  for uc in $(oc get upgradeconfig -n openshift-managed-upgrade-operator osd-upgrade-config -o name --ignore-not-found); do
+                     oc delete upgradeconfig -n openshift-managed-upgrade-operator --ignore-not-found=true managed-upgrade-config 
```

Comment:
why do we try to remove the upgrade config with new name `managed-upgrade-config` here? and recreate it in next step?

Reply:
This was mainly because I wanted to always treat "osd-upgrade-config" as the source of truth to replace with a "managed-" equivalent, and not whatever "managed-" might be floating around at the same time. This could potentially be important if the cluster happens to be mid-upgrade to preserve what is in its `Status` (although I'd _hope_ that a mid-upgrade cluster doesn't end up with a "managed-" UC). 

Once an existing `managed-*` has been deleted and replaced with the contents of `osd-upgrade-config` and `osd-upgrade-config` the job should not attempt to re-delete on a re-run because `osd-upgrade-config` will no longer exist, immediately short-circuiting the loop.